sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/BaseHibernatePaginatedDataSource.java

```diff
@@ -115,10 +110,30 @@ default Criteria createPaginationCriteria() throws IOException {
    */
   Criterion searchRestrictions(String query);
 
-  /**
-   * Add additional restrictions to the {@link Criteria} object. Note, during listing, a collection of IDs is fetched and then these are
-   * used to perform a second query. In that instance, this method is <b>not</b> called for the ID-driven fetch.
-   */
-  default void setAdditionalPaginationCriteria(Filter filter, Criteria criteria) {
+  @Override
+  default void setFulfilled(Criteria item, boolean isFulfilled) {
+    throw new IllegalArgumentException();
```

Comment:
I see some of your points, but they can be addressed while still having a separate filter class for each entity type which have their unique fields separate from the rest.

I'm not sure what you mean about trading compile time errors for runtime since the former should always be preferable. If you mean checked vs. unchecked Exceptions, that's not what I'm talking about. What I mean is that it should not be possible to set a "fulfilled" property on a filter for Samples since that property is only applicable to Orders.